### ISSUE TITLE
Choose a database & layer for storing a local copy of the blockchain

### ISSUE THREAD

**Issue by ndane** [Blockchainers/core#7]

**Comment by ndane** [comment]
Code for this will be put in src/core/db

**Comment by HardyHardy08** [comment]
Nathan, I'm still a bit confused on how to properly complete this. Were I to choose a relational database (say MySQL or PostgreSQL) and find a library to interface it in C++. Will that be OK? I'm not sure if there're any other aspects I should be considering. SQL databases are pretty quick, no?

**Comment by ndane** [comment]
@HardyHardy08 Yeah, I think the best thing to do is go and find a database which

- [ ] Sounds good
- [ ] Works with the Apache 2.0 License
- [ ] Is efficient and small
- [ ] Is easy to use in C++

And then create a new branch demoing how it can be used in the project so everyone can discuss it further.


I'm doing a similar thing with the unit testing task at the moment, trying to find the best tool for the job.
You can see the progress on my branch here: https://github.com/Blockchainers/core/tree/feature/catch-demo

**Comment by HardyHardy08** [comment]
Cool! Roger that. Let me rediscover how C++ works again first ahha
